addons/payment/views/payment_templates.xml

```diff
@@ -368,27 +368,50 @@
         <!-- Variables description:
             - 'tx' - The transaction whose status must be displayed
         -->
-        <div t-if="tx.state == 'pending' and not is_html_empty(tx.acquirer_id.sudo().pending_msg)"
-             class="alert alert-warning alert-dismissible">
-            <span t-out="tx.acquirer_id.sudo().pending_msg"/>
-            <button class="btn-close" data-bs-dismiss="alert" title="Dismiss"></button>
-        </div>
-        <div t-elif="tx.state == 'authorized' and not is_html_empty(tx.acquirer_id.sudo().auth_msg)"
-             class="alert alert-success alert-dismissible">
-            <span t-out="tx.acquirer_id.sudo().auth_msg"/>
-            <button class="btn-close" data-bs-dismiss="alert" title="Dismiss"></button>
-        </div>
-        <div t-elif="tx.state == 'done' and not is_html_empty(tx.acquirer_id.sudo().done_msg)"
-             class="alert alert-success alert-dismissible">
-            <span t-out="tx.acquirer_id.sudo().done_msg"/>
-            <button class="btn-close" data-bs-dismiss="alert" title="Dismiss"></button>
-        </div>
-        <div t-elif="tx.state == 'cancel' and not is_html_empty(tx.acquirer_id.sudo().cancel_msg)"
-             class="alert alert-danger alert-dismissible">
-            <span t-out="tx.acquirer_id.sudo().cancel_msg"/>
-            <button class="btn-close" data-bs-dismiss="alert" title="Dismiss"></button>
+        <t t-if="tx.state == 'draft'">
+            <t t-set="alert_style">info</t>
+            <t t-set="status_message">
+                <p>Your payment has not been processed yet.</p>
+            </t>
+        </t>
+        <t t-elif="tx.state == 'pending'">
+            <t t-set="alert_style">warning</t>
+            <t t-set="status_message"
+               t-value="tx.acquirer_id.sudo().pending_msg"
+               t-if="not is_html_empty(tx.acquirer_id.sudo().pending_msg)"/>
+        </t>
+        <t t-elif="tx.state == 'authorized'">
+            <t t-set="alert_style">success</t>
+            <t t-set="status_message"
+               t-value="tx.acquirer_id.sudo().auth_msg"
+               t-if="not is_html_empty(tx.acquirer_id.sudo().auth_msg)"/>
+        </t>
+        <t t-elif="tx.state == 'done'">
+            <t t-set="alert_style">success</t>
+            <t t-set="status_message"
+               t-value="tx.acquirer_id.sudo().done_msg"
+               t-if="not is_html_empty(tx.acquirer_id.sudo().done_msg)"/>
+        </t>
+        <t t-elif="tx.state == 'cancel'">
+            <t t-set="alert_style">danger</t>
+            <t t-set="status_message"
+               t-value="tx.acquirer_id.sudo().cancel_msg"
+               t-if="not is_html_empty(tx.acquirer_id.sudo().cancel_msg)"/>
+        </t>
+        <t t-elif="tx.state == 'error'">
+            <t t-set="alert_style">danger</t>
+            <t t-set="status_message">
+                <p>An error occurred during the processing of your payment.</p>
+            </t>
+        </t>
```

Comment:
Or two methods in python to simplify the xml?
```xml
<t t-set="status_message" t-value="tx._get_status_message()"/>
<t t-set="status_style" t-value="tx._get_status_message_style()"/>
```

Reply:
This code is really QWeb-related so I don't really mind if it takes a bit more space in a QWeb-template^^